op-challenger/game/fault/trace/vm/executor.go

```diff
@@ -33,22 +33,28 @@ type Config struct {
 	InfoFreq         uint // Frequency of progress log messages (in VM instructions)
 }
 
+type VmArgs interface {
+	Cfg() Config
+	LocalInputs() *utils.LocalGameInputs
+	SetLocalInputs(utils.LocalGameInputs)
+	FillHostCommand(args *[]string, dataDir string) error
+}
+
 type Executor struct {
-	cfg              Config
+	VmArgs
```

Comment:
Personally I'd say the executor _has_ a VmArgs rather than it _being_ a VmArgs so I'd make this 
```suggestion
	vmArgs VmArgs
```

Reply:
Is there a semantic difference here? Struct embedding just allows us to short-hand; I don't believe this changes any functionality.

Reply:
It changes what the exposed API is and the meaning of the design.  Embedding the struct is saying that `Executor` is everything that a `VmArgs` is plus some extra stuff.  But really `Executor` is a thing that uses a `VmArgs` to help it do its job. You're right that it works either way, but making the design side clear makes it easier to see the separation of responsibilities - `executor.ServerArgs` makes it look like the executor is responsible for the server, whereas `executor.vmArgs.ServerArgs` makes it clear that responsibility is being delegated elsewhere (without having to jump back up to the struct definition to discover it).